src/ga/individual.rs

```diff
@@ -1,43 +1,41 @@
 use std::fmt::Debug;
-
 use serde::Serialize;
 
-pub trait Gene: Sized + Default + Copy + Debug {}
 
-// Blanket implementaion.
-impl<T: Sized + Default + Copy + Debug> Gene for T {}
+pub trait Chromosome: Sized + Sync + Send + Clone + Default + Debug {}
 
-pub type Chromosome<T> = Vec<T>;
+/// Blanket implementation
+impl<T: Sized + Sync + Send + Clone + Default + Debug> Chromosome for T {}
 
-pub trait ChromosomeWrapper<T: Gene>: Ord + Clone + Debug {
+pub trait ChromosomeWrapper<T: Chromosome>: Sized + Sync + Clone + Ord + Debug {
```

Comment:
Trait bound `Ord` here is quite problematic. It is required for `sort` algorithm and it is not implemented by default for `f64` type, thus working with solutions represented by e.g. `Vec<f64>` requires manually implementing `Ord` -- and [it is not trivial to do it right](https://users.rust-lang.org/t/how-to-sort-a-vec-of-floats/2838/2). For now I provided mock (although working) implementation. You can see it in `individual` module.